tests/data/image_without_release.yaml

```diff
@@ -0,0 +1,37 @@
+version: 1
+
+upload:
+  - source: "canonical/rocks-toolbox"
+    commit: 17916dd5de270e61a6a3fd3f4661a6413a50fd6f
+    directory: mock_rock/1.0
+    release:
+      1.0-22.04:
+        end-of-life: "2024-05-01T00:00:00Z"
+        risks:
+          - candidate
+          - edge
+          - beta
+  - source: "canonical/rocks-toolbox"
+    commit: 17916dd5de270e61a6a3fd3f4661a6413a50fd6f
+    directory: mock_rock/1.1
+    release:
+      1.1-22.04:
+        end-of-life: "2030-05-01T00:00:00Z"
+        risks:
+          - candidate
+          - edge
+          - beta
+      1-22.04:
+        end-of-life: "2030-05-01T00:00:00Z"
+        risks:
+          - candidate
+          - edge
+          - beta
+  - source: "canonical/rocks-toolbox"
+    commit: 17916dd5de270e61a6a3fd3f4661a6413a50fd6f
+    directory: mock_rock/1.2
+    release:
```

Comment:
as per my comment above, you may want to have one of these tests have no `release` fields at all, since those are optional

Reply:
Makes sense. I added additional tests for the image_trigger file which should cover most of the cases.